tests/internal/test_packages.py

```diff
@@ -84,3 +84,5 @@ def test_filename_to_package():
 
     package = packages.filename_to_package(gp.__file__)
     assert package is None or package.name == "protobuf"
+
+    del packages._package_file_mapping.__closure__[0].cell_contents.__callonce_result__
```

Comment:
perhaps we can change the way the `callonce` decorator is implemented. Instead of returning a function, we can return an instance of a class that stores the computed value, but also offers a method to clear the cache?